lib/Backends/CPU/CPUFactory.cpp

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CPUBackend.h"
+
+namespace glow {
+
+/// Register CPU backend via CPUFactory.
+class CPUFactory : public BaseFactory<BackendKind, Backend> {
```

Comment:
Should the Factory class definition be a macro as well? It seems like all these Factories are pretty much the same. Only their names and their BackendKind differ.